Deploy

MajoBerger opened the pull request:
| Phases            | MM  |  MB  | MR |   JM | Total  |
|-----------------|----:|----:|-----:|-----:|-------:|
| ETA                  |  0  |    - |     0 |      4 |        4 |
| Developing      |  0  |   3 |    0 |      0 |         3 |
| Review             |  0  |    0 |    0 |      0 |         0 |
| Total                |   -  |   -   |  -    |   -    |         7 |
| ETA est.             |      |       |       |         |         0 |
| ETA cust.           |   -  |   -  |   -   |   -     |        0 |
## Deployment scripts
Re-used docker for our purpose.

start.*bat starts everything and creates admin account, similarly the other tasks.
Images in docker compose are taken from our company docker hub https://hub.docker.com/u/dataquest

MajoBerger committed:
adding support for automatic deploy

MajoBerger committed:
removed build section; added redeploy scripts

vidiecan commented on docker/cli.yml:
```diff
@@ -16,7 +16,7 @@ version: "3.7"
 services:
   dspace-cli:
-    image: "${DOCKER_OWNER:-dspace}/dspace-cli:${DSPACE_VER:-dspace-7_x}"
+    image: "${DOCKER_OWNER:-dataquest}/dspace-cli:${DSPACE_VER:-dspace-7_x}"
```
why? 
1) change to an upstream file
2) we can control it using env so why change it?

vidiecan commented on docker/deploy.bat:
```diff
@@ -0,0 +1,2 @@
+docker-compose -p our_dspace -f deploy.yml up -d
```
I would rather create "our" directory with scripts e.g., we use
`build-scripts/build` `build-scripts/test` etc.

MajoBerger replied:
used build-scripts

vidiecan commented on docker/deploy.bat:
```diff
@@ -0,0 +1,2 @@
+docker-compose -p our_dspace -f deploy.yml up -d
+docker-compose -p our_dspace -f cli.yml run --rm dspace-cli create-administrator -e test@test.edu -f admin -l user -p admin -c en
```
I use `pause` at the end of batch so I see what happend once finished

MajoBerger replied:
added

MajoBerger committed:
moved scripts to build-scripts/deploy-scripts; reverted docker/cli.yml to upstream version; used .env file

MajoBerger committed:
added pulling image before deploy and redeploy; using tag ..7_x instead of ..7_x-test

MajoBerger committed:
First deploynment touch

MajoBerger committed:
first script run

MajoBerger committed:
script run 0

MajoBerger committed:
kill useless runs

MajoBerger committed:
deploy run 1

MajoBerger committed:
deploy run 2

MajoBerger committed:
deploy run 3

MajoBerger committed:
deploy run 4

MajoBerger committed:
deploy run 5

MajoBerger committed:
deploy run 6

MajoBerger committed:
deploy run 7

MajoBerger committed:
deploy run 8

MajoBerger committed:
deploy run 9

MajoBerger committed:
deploy run 10 (this was lame)

MajoBerger committed:
= instead of : in .env file

vidiecan commented on docker/docker-compose.yml:
```diff
@@ -24,7 +24,7 @@ services:
       DSPACE_REST_PORT: 8080
       DSPACE_REST_NAMESPACE: /server
-    image: dspace/dspace-angular:dspace-7_x
+    image: dataquest/dspace-angular:dspace-7_x
```
is this necessary?

vidiecan commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,25 @@
+  deploy:
+    runs-on: dspace-dep-1
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
```
I do not think the GITHUB_WORKSPACE is required, see https://github.com/actions/checkout

vidiecan commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,25 @@
+
+      # Ready script
+      - name: Ready script
+        run: chmod u+x $GITHUB_WORKSPACE/build-scripts/deploy-scripts/deploy.sh
```
`+x` change in git
https://stackoverflow.com/questions/21691202/how-to-create-file-execute-mode-permissions-in-git-on-windows
git update-index --chmod=+x foo.sh

vidiecan commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,25 @@
+      - name: change, list and run
+        run: |
+          cd $GITHUB_WORKSPACE/build-scripts/deploy-scripts/
+          pwd
```
this is not crucial, think about using
`pwd || true`

vidiecan commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,25 @@
+      - name: Ready script
+        run: chmod u+x $GITHUB_WORKSPACE/build-scripts/deploy-scripts/deploy.sh
+
+      - name: change, list and run
```
update name

vidiecan commented on .github/workflows/deploy.yml:
```diff
@@ -0,0 +1,25 @@
+
+jobs:
+  deploy:
+    runs-on: dspace-dep-1
```
add comment on what this is

MajoBerger committed:
changed rest_host

MajoBerger committed:
rest_host update

MajoBerger committed:
reverted rest_host

MajoBerger committed:
dev5pc

MajoBerger committed:
deploy hotfix n0

MajoBerger committed:
Merge branch 'deploy' of github.com:dataquest-dev/dspace-angular into deploy

MajoBerger committed:
deploy hotfix n1

MajoBerger committed:
deploy hotfix n2

MajoBerger committed:
deploy hotfix n3

MajoBerger committed:
deploy hotfix n4

MajoBerger committed:
deploy hotfix n5

MajoBerger committed:
deploy hotfix n6

MajoBerger committed:
tunnel reset attempt 0

milanmajchrak committed:
Update deploy.yml

MajoBerger committed:
move to /dspace

MajoBerger committed:
try fix

MajoBerger committed:
revert

MajoBerger committed:
remove slash

MajoBerger committed:
moved to /dspace again

MajoBerger committed:
last

MajoBerger committed:
Update deploy.yml

MajoBerger committed:
both slashes

MajoBerger committed:
remain at /

MajoBerger committed:
relocate server

MajoBerger committed:
revert server to 8080

vidiecan committed:
[devOps]
* reuse docker* files from upstream, make them more configurable
* described docker in build-scripts/
* TODO REVERT: try without IPAM
* TODO REVERT: disable host check

fixes #31 by adding env. variables into docker compose

Identified problems
- login does not work because of host/origin check (very likely)

Commit:
[devops] deploy

vidiecan committed:
Merge branch 'dtq-dev' into deploy
